### PR TITLE
Add `@RestrictedSince` to update-since-todo.sh

### PR DESCRIPTION
**Pull request by NotMyFault**
This PR amends my previous PR https://github.com/jenkinsci/jenkins/pull/6396 which addresses left over `@since TODO`s from past releases. Unfortunately, `@RestrictedSince("TODO")`s weren't considered by it.

The change proposed does take care of it. To verify my changes, I ran the tool on the master branch: https://gist.github.com/NotMyFault/81ff02b02875eec1f0cf72440fe8824a

### Proposed changelog entries

* N/A, skip-changelog

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [X] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change) and are in the imperative mood. [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade

### Desired reviewers

@mention

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are accurate, human-readable, and in the imperative mood
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins-ci.org/issues/?filter=12146)).

### CODE REVIEW

**Review by basil — APPROVED** [review]

**Review by timja — APPROVED** [review]

**Comment by timja** [comment]
This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!

**Review by daniel-beck — CHANGES_REQUESTED** [review]
The console output in the gist that was added to make reviews of PRs easier if that's provided (e.g. #6331) is now misleading:

> List of commits introducing new API and the first release they went in:

Could this be split into the existing output + a separate section for `@RestrictedSince`?

**Comment by NotMyFault** [comment]
I've changed the wording a bit and moved the output.

**Comment by alecharp** [comment]
Removing the label until Daniel had a chance to review the recent changes.

**Comment by alecharp** [comment]
@daniel-beck do you have some time to spend on this PR? Thanks.

**Comment by basil** [comment]
> Could this be split into the existing output + a separate section for `@RestrictedSince`?

Filing this under the category of "easier said than done," especially in the current shell script. Though could be done much more easily after #6458.

**Comment by NotMyFault** [comment]
Closing in favor of basil's approach <3

**Comment by basil** [comment]
> Closing in favor of basil's approach.

My approach to what, specifically? You must not be referring to my approach to adding `@RestrictedSince` support to our update script, because I have never done that in any PR, nor indicated an intention to. The idea in #6458 was for you to merge in my rewrite into this PR, accommodate Daniel's preferences by splitting the output into multiple sections (which would then be much easier), then to finally merge this PR.

**Pull request closed without merging**
